evalml/pipelines/components/estimators/classifiers/kneighbors_classifier.py

```diff
@@ -0,0 +1,50 @@
+import numpy as np
+from sklearn.neighbors import KNeighborsClassifier as SKKNeighborsClassifier
+from skopt.space import Integer
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+
+
+class KNeighborsClassifier(Estimator):
+    """
+    K-Nearest Neighbors Classifier.
+    """
+    name = "K-Nearest Neighbors Classifier"
+    hyperparameter_ranges = {
+        "n_neighbors": Integer(2, 12),
```

Comment:
I wonder if there's a different way to parametrize this? I imagine that for some datasets, the estimator might need more than 12 neighbors to get good performance.

Maybe we can parametrize this as the percentage of the training data used in the neighborhood. This is not blocking merge, I'm just thinking out loud hehe.